File: March_2024/D8/Solution.java
package March_2024.D8;

public class Solution {
    public static void main(String[] args) {

    }

    public int maxFrequencyElements(int[] nums) {
        int maxValue = Integer.MIN_VALUE;
        for (int num : nums) {
            if (maxValue < num) {
                maxValue = num;
            }
        }
        int arr[] = new int[maxValue + 1];
        for (int i = 0; i < nums.length; i++) {
            arr[nums[i]]++;
        }
        int maxFreq = 0;
        for (int val : arr) {
            if (val > maxFreq) {
                maxFreq = val;
            }
        }
        int count = 0;
        for (int val : arr) {
            if (val == maxFreq) {
                count++;
            }
        }
        return count * maxFreq;
    }
}

File: March_2024/D8/Solution.py
class Solution(object):
    def maxFrequencyElements(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        frequency_dict = {}
        for i in nums:
            if i not in frequency_dict:
                frequency_dict[i] = 1
            else:
                frequency_dict[i] += 1
        MaxFrequency = max(frequency_dict.items(), key = operator.itemgetter(1))[1]
        count = 0
        for i in frequency_dict:
            if frequency_dict[i] == MaxFrequency:
                count += MaxFrequency
        return count